From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Dominik Csapak <d.csapak@proxmox.com>
Date: Wed, 9 May 2018 10:47:53 +0200
Subject: [PATCH] add custom fbresize event on rfb

this can be use to react to changes of the resolution, like resizing the
window

Signed-off-by: Dominik Csapak <d.csapak@proxmox.com>
---
 app/ui.js   | 11 +++++++++++
 core/rfb.js | 10 ++++++++++
 2 files changed, 21 insertions(+)

diff --git a/app/ui.js b/app/ui.js
index bd1561c..3726c8c 100644
--- a/app/ui.js
+++ b/app/ui.js
@@ -1116,6 +1116,7 @@ const UI = {
         UI.rfb.addEventListener("clipboard", UI.clipboardReceive);
         UI.rfb.addEventListener("bell", UI.bell);
         UI.rfb.addEventListener("desktopname", UI.updateDesktopName);
+        UI.rfb.addEventListener("fbresize", UI.updateSessionSize);
         UI.rfb.clipViewport = UI.getSetting('view_clip');
         UI.rfb.scaleViewport = UI.getSetting('resize') === 'scale';
         UI.rfb.resizeSession = UI.getSetting('resize') === 'remote';
@@ -1775,6 +1776,16 @@ const UI = {
         document.getElementById('pve_commands_button').classList.remove("noVNC_selected");
     },
 
+    updateSessionSize: function(e) {
+        var rfb = e.detail.rfb;
+        var width = e.detail.width;
+        var height = e.detail.height;
+        UI.PVE.updateFBSize(rfb, width, height);
+
+        UI.applyResizeMode();
+        UI.updateViewClip();
+    },
+
 /* ------^-------
  *    /PVE
  * ==============
diff --git a/core/rfb.js b/core/rfb.js
index e3266cc..23b436c 100644
--- a/core/rfb.js
+++ b/core/rfb.js
@@ -3014,6 +3014,16 @@ export default class RFB extends EventTargetMixin {
         this._updateClip();
         this._updateScale();
 
+        // fbresize event
+        var event = new CustomEvent("fbresize", {
+            detail: {
+                rfb: this,
+                width: width,
+                height: height }
+            }
+        );
+        this.dispatchEvent(event);
+
         this._updateContinuousUpdates();
 
         // Keep this size until browser client size changes
-- 
2.39.5

